libact/base/dataset.py

```diff
@@ -142,44 +171,42 @@ def format_sklearn(self):
         y : numpy array, shape = (n_samples)
             Sample labels.
         """
-        X, y = zip(*self.get_labeled_entries())
-        return np.array(X), np.array(y)
+        # becomes the same as get_labled_entries
+        X, y = self.get_labeled_entries()
+        return X, np.array(y)
 
     def get_entries(self):
         """
         Return the list of all sample feature and ground truth tuple.
 
         Returns
         -------
-        data : list, shape = (n_samples)
-            List of all sample feature and label tuple.
+        X: numpy array or scipy matrix, shape = ( n_sample, n_features )
+        y: numpy array, shape = (n_samples)
         """
-        return self.data
+        return self._X, self._y
 
     def get_labeled_entries(self):
         """
         Returns list of labeled feature and their label
 
         Returns
         -------
-        labeled_entries : list of (feature, label) tuple
-            Labeled entries
+        X: numpy array or scipy matrix, shape = ( n_sample labeled, n_features )
+        y: list, shape = (n_samples lebaled)
         """
-        return list(filter(lambda entry: entry[1] is not None, self.data))
+        return self._X[self.get_labeled_mask()], self._y[self.get_labeled_mask()].tolist()
```

Comment:
Do we need to make y into a list, can we just keep it numpy array?

Reply:
We actually need this. 
For some multi-label code, I think they are taking advantage of the nested structure of out output. I removed it ran against the unit test and got back with a lot of fails. So I would suggest just keep it as a list. And I don't think the performance would not be affected too much.

Reply:
Ok, I think we can merge first and I'll look into this later.

Reply:
Cool. I am starting to adapt libact to my own research experiments. So might start opening more full requests for improvements in the future :)